### ISSUE TITLE
Show warning when git is not initialized

### ISSUE THREAD

**Issue by vytautas-pranskunas-** [therealklanni/guppy-cli#39]
When installing guppy-pre commit (not surr about others) show warning message if git folder is missing instead of hanging whole npm i process

**Comment by therealklanni** [comment]
Did you try `npm i --production` like I suggested?

**Comment by vytautas-pranskunas-** [comment]
I left you comment regarding this in another post

https://github.com/therealklanni/git-guppy/issues/82#issuecomment-232739771

**Comment by therealklanni** [comment]
You said:

> Hi, i cannot do that because on CI is building dev env and running karma tests which are under dev dependency same as guppy.

I don't think this is necessarily an issue with guppy. It seems to me more like an issue with how your CI is set up. You should probably check out the branch in your CI, rather than (what I assume you're doing is) copying your codebase to CI. That way you have the `.git` folder structure. Unfortunately, guppy cannot run without that. Your other option is to set up your CI to remove the guppy dependency when CI starts it's build/run process if you don't want guppy in the CI at all (not the best/easiest approach).

The first option is probably the best approach, because then you can simply write your gulp `pre-commit` (and whatever other hooks you're using) to skip if the `NODE_ENV` is "staging" or whatever you call the CI env.

I'm just not fully convinced the responsibility is on guppy.

**Comment by vytautas-pranskunas-** [comment]
So you claim that plugin rather then throw warning should hang whole process?

**Comment by therealklanni** [comment]
No, I'm telling you that it seems like your CI is not properly set up. If it were properly set up, you wouldn't have an issue.

**Comment by vytautas-pranskunas-** [comment]
Can you fix that it would not hang the process because it is a matter of one if exists check, and we will improve our build process?

**Comment by therealklanni** [comment]
I can look into it and see if it's a useful change. It's going to be pretty low priority, though, considering it's easy to resolve with a proper CI setup.

**Comment by vytautas-pranskunas-** [comment]
I know what you mean but in large corporations where you do not have rights to influence build process such simple lib change can safe loads of time while banging head to get permissions or describe problems to devops. So i would be really greatfull if you add that 5mins fix.

**Comment by therealklanni** [comment]
You're welcome to submit a PR for review. Unfortunately, I have a lot of other things that are higher priority for me, so I won't be able to give this attention right away. I'm sorry.

**Comment by vytautas-pranskunas-** [comment]
I think i would not have time for PR, if devops wpuld not change build way i will have to look fol alternative lib. However i think it is good developement practice to make isolated libs to not break whole chain 😏

**Comment by therealklanni** [comment]
Again, sorry you're having issues with your build. You keep asserting that _guppy_ is breaking your build, unfortunately I don't agree with that assertion. It sounds like your build is not correctly configured. I don't mind adding it as an enhancement, but it's pretty low priority considering you're the only one requesting the enhancement. Sorry if it sounds insensitive to your situation, it's just that I have a lot of other things on my plate that are more urgent.

If you or someone else can work on this in a more timely manner than I can, I would definitely appreciate the contribution.

**Comment by jachin84** [comment]
This issue also broke our CI build. If you are using TeamCity with the built-in git configuration the .git folder is not copied to the agents working directory. You can fix the issue by installing the latest version of Git which does seem to create the .git folder in the working directory.

**Comment by kane-mason** [comment]
also getting this issue
